src/big_core_input.cpp

```diff
@@ -112,13 +119,15 @@ const T& BigCoreInput::at(uint64_t imageNum, uint64_t row, uint64_t column, uint
 	if (tileNum >= this->numberOfTiles)
 		throw "Tile number out of bound!";
 
+	int64_t offset = this->outermostEntitiesOffsets[imageNum];
+
 	uint64_t dataTypeIndex = this->isUniformDataType() ? 0 : imageNum;
 	if (this->isInMemory())
 	{
-		return static_cast<const T>(this->data[this->outermostEntitiesOffsets[imageNum] + row * column * plane * tileNum * this->getImageType(this->dataType[dataTypeIndex])]);
+		return static_cast<const T>(this->data[offset + row * column * plane * tileNum * this->getImageType(this->dataType[dataTypeIndex])]);
 	}
 
-	this->file.seekg(this->dataPosition + this->outermostEntitiesOffsets[imageNum]);
+	this->file.seekg(this->dataPosition + offset);
 	uint64_t size = this->outermostEntitiesOffsets[imageNum + 1] - this->outermostEntitiesOffsets[imageNum];
 	char* ret = new char[size];
 	this->file.read(ret, size);
```

Comment:
Na variantu, že se data nevejdou do paměti, se zatím vykašli. Zbytečně to odvádí pozornost od opravdových problémů (což je hlavně indexování) a může se to dodělat až později, až bude fungovat varianta, že se všechno do paměti vejde.